modules/git/repo_compare.go

```diff
@@ -267,3 +274,57 @@ func (repo *Repository) GetDiffFromMergeBase(base, head string, w io.Writer) err
 	}
 	return err
 }
+
+// ReadPullHead will fetch a pull ref if possible or return an error
+func (repo *Repository) ReadPullHead(prID int64) (commitSHA string, err error) {
+	headPath := fmt.Sprintf("refs/pull/%d/head", prID)
+	fullHeadPath := filepath.Join(repo.Path, headPath)
+	loadHead, err := os.Open(fullHeadPath)
+	if err != nil {
+		return "", err
+	}
+	defer loadHead.Close()
+	// Read only the first line of the patch - usually it contains the first commit made in patch
+	scanner := bufio.NewScanner(loadHead)
+	scanner.Scan()
+	commitHead := scanner.Text()
+	if len(commitHead) != 40 {
+		return "", errors.New("head file doesn't contain valid commit ID")
+	}
+	return commitHead, nil
+}
```

Comment:
NO!

This will break as soon as references are packed into files! 

We have git.IsReferenceExist, we have (*git.Repository).IsReferenceExist 

You should use the functionality that we have already written for this.

Reply:
This should have been:

```go
	headRef := fmt.Sprintf(PullPrefix+"%d/head", prID)
	return repo.GetRefCommitID(headRef)
```